spec/types/date.spec.ts

```diff
@@ -0,0 +1,16 @@
+import { z } from 'zod';
+import { expectSchema } from '../lib/helpers';
+
+describe('date', () => {
+  it('supports dates and sets the type to "string"', () => {
+    const example = new Date('2022-10-14T09:39:30Z');
+    const schema = z.date().openapi({ refId: 'Date', example });
+
+    expectSchema([schema], {
+      Date: {
+        type: 'string',
+        example,
```

Comment:
Since it's a type: string the example should probably be `example: example.toISOString()` instead of the date object itself. This way we're not relying on the yaml or json serializer to decide how to display it for us.

Reply:
Thank you for the suggestion, I'm now serializing examples in the generator to ISO date strings. Is this what you meant?

Reply:
Maybe this is too opinionated because not everyone might want to use ISO date strings. Instead we could change the example TS type from Date to string when the schema is a ZodDate (or derive the example type from the openapi metadata type). What do you think?

Reply:
I think that the example should be defined by the user and not zod-to-openapi. But yes, it should be a string (Typescript wise)

Reply:
Ok I changed the example type to `string` when the schema type infers to `Date`. I also reverted the automatically generated example so users can define them.